src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js

```diff
@@ -0,0 +1,57 @@
+(function () {
+    "use strict";
+
+    function PartialViewMacrosCreateController($scope, codefileResource, $location, navigationService) {
+
+        var vm = this;
+        var node = $scope.dialogOptions.currentNode;
+
+        vm.snippets = [];
+        vm.showSnippets = false;
+        vm.creatingFolder = false;
+
+        vm.createPartialViewMacro = createPartialViewMacro;
+        vm.showCreateFolder = showCreateFolder;
+        vm.createFolder = createFolder;
+        vm.showCreateFromSnippet = showCreateFromSnippet;
+
+        function onInit() {
+            codefileResource.getSnippets('partialViewMacros')
+                .then(function(snippets) {
+                    vm.snippets = snippets;
+                });
+        }
+
+        function createPartialViewMacro(selectedSnippet) {
+
+            var snippet = null;
+
+            if(selectedSnippet && selectedSnippet.fileName) {
+                snippet = selectedSnippet.fileName;
+            }
+
+            $location.search('create', null);
```

Comment:
@madsrasmussen I am not sure I understand this code here?
You are setting the create & snippet part of the URL to null , then changing the url/redirecting to the file editor & then re-adding back on the create & search snippet stuff.